x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/index.ts

```diff
@@ -345,13 +358,25 @@ export class AIAssistantKnowledgeBaseDataClient extends AIAssistantDataClient {
    */
   public isESQLDocsLoaded = async (): Promise<boolean> => {
     const esqlDocs = await this.getKnowledgeBaseDocumentEntries({
-      query: ESQL_DOCS_LOADED_QUERY,
-      // kbResource, // Note: `8.15` installs have kbResource as `unknown`, so don't filter yet
+      query: '',
+      kbResource: ESQL_RESOURCE,
```

Comment:
This might be an issue for existing serverless releases as there was a time (I think just `8.15.x`) where ESQL docs were loaded as `kbResouce: 'unknown'`. So this would return false and the assistant would forever show the 'Setup KB' button.

I should've just made a `isESQLDocsLoadedLegacy` variant and used that when the FF is disabled in the get KB status route. I think we might wanna do that now to ensure we're not introducing this setup 'bug' for serverless installs on upgrade.